1_fetch/src/fetch_nwis.R

```diff
@@ -30,14 +30,110 @@ fetch_gw_uv <- function(target_name, gw_sites, start_date, end_date, param_cd) {
     write_feather(target_name)
 }
 
-convert_uv_to_dv <- function(target_name, gw_uv_data_fn) {
+convert_uv_to_dv <- function(target_name, gw_uv_data_fn, site_tz_xwalk) {
   read_feather(gw_uv_data_fn) %>% 
-    mutate(Date = as.Date(dateTime)) %>% 
+    
+    ### Convert to date using local timezone
+    
+    # Remove the tz_cd column that is downloaded with the data
+    select(-tz_cd) %>% 
+    # Join in the local timezone information for each site
+    left_join(site_tz_xwalk, by = "site_no") %>% 
+    # Change the dateTime column to just the date. Need to use the timezone
+    # rather than UTC so that days are treated appropriately. Seems overly 
+    # complicated but you can't use `format()` with more than one tz, so 
+    # this vectorizes that call per timezone. Not using rowwise() because 
+    # that would be slow. Doing a single call per timezone speeds this up.
+    group_by(tz_cd) %>% 
+    nest() %>% 
+    pmap(function(tz_cd, data) {
+      data %>% mutate(Date = POSIXct_to_Date_tz(dateTime, tz_cd))
+    }) %>% 
+    bind_rows() %>% 
+    
+    ### Reduce each instantaneous value to a single average for each date
     group_by(site_no, Date) %>% 
     summarize(GWL = mean(GWL_inst, na.rm = TRUE)) %>% 
     write_feather(target_name)
 }
 
+# Convert POSIXct to Dates for a given timezone
+# Only works with one tz_abbr at a time
+POSIXct_to_Date_tz <- function(posix_dates, tz_abbr) {
```

Comment:
These next two function are duplicated in `0_historic/src/fetch_nwis_historic`

Reply:
I'm not sure I understand the need for doing that. I know we're keeping the historic fetch separate from the current, but seems like duplicating the same steps increases complexity and room for error